dotnet/src/Connectors/Connectors.AzureOpenAI/Core/ClientCore.File.cs

```diff
@@ -96,28 +146,48 @@ internal async Task<IEnumerable<OpenAIFileReference>> GetFilesAsync(Cancellation
         return response.Value.ToArray();
     }
 
-    private static OpenAIFileReference ConvertToFileReference(OpenAIFileInfo fileInfo)
+    private static AzureOpenAIFileReference ConvertToFileReference(OpenAIFileInfo fileInfo)
         => new()
         {
             Id = fileInfo.Id,
             CreatedTimestamp = fileInfo.CreatedAt.DateTime,
             FileName = fileInfo.Filename,
             SizeInBytes = (int)(fileInfo.SizeInBytes ?? 0),
-            Purpose = ConvertToFilePurpose(fileInfo.Purpose),
+            Purpose = ConvertToAzureOpenAIFilePurpose(fileInfo.Purpose),
         };
 
-    private static FileUploadPurpose ConvertToOpenAIFilePurpose(SKFilePurpose purpose)
+    private static FileUploadPurpose ConvertToOpenAIFileUploadPurpose(AzureOpenAIFileUploadPurpose purpose)
```

Comment:
Since our abstractions are enums we can use the `switch expression`. Only when we try to convert to our type we are forced to use if statements.

Reply:
It's not relevant anymore because the conversion code has been dropped.